src/main/java/com/okta/developer/web/rest/AccountResource.java

```diff
@@ -39,21 +39,19 @@ public AccountResource(UserService userService) {
     /**
      * {@code GET  /account} : get the current user.
      *
+     * @param principal the current user; resolves to {@code null} if not authenticated.
      * @return the current user.
      * @throws AccountResourceException {@code 500 (Internal Server Error)} if the user couldn't be returned.
      */
     @GetMapping("/account")
-    public AdminUserDTO getAccount() {
-        Optional<String> email = SecurityUtils.getCurrentUserLogin();
-
-        if (email.isPresent()) {
-            Optional<User> user = userService.getUserWithAuthoritiesByLogin(email.get());
-            if (user.isPresent()) {
-                return new AdminUserDTO(user.get());
-            }
+    @SuppressWarnings("unchecked")
+    public AdminUserDTO getAccount(Principal principal) {
+        Optional<User> user = userService.getUserWithAuthoritiesByLogin(principal.getName());
+        if (user.isPresent()) {
+            return new AdminUserDTO(user.get());
+        } else {
+            throw new AccountResourceException("User could not be found");
```

Comment:
Reverting to the previous version, it no longer needs the email address.

That said, the JHipster login now shows:
> You are logged in as user: `<sub claim>`

I'm not sure if that is expected or not 🤷